### PR TITLE
Add custom controller documentation

### PR DESCRIPTION
**Pull request by esaldiran**
Custom controller support 2nd version wiki https://github.com/ArduPilot/ardupilot/pull/21216

### CODE REVIEW

File: dev/source/docs/copter-adding-custom-controller.rst
Comment (Hwurzburg) [inline dev/source/docs/copter-adding-custom-controller.rst:23]:
```suggestion
- Custom controller parameters start with ``CC_`` in the GCS.
```
to avoid a build error...thinks its a param link

File: dev/source/docs/copter-adding-custom-controller.rst
Comment (Hwurzburg) [inline dev/source/docs/copter-adding-custom-controller.rst:29]:
```suggestion
- :ref:`CC_TYPE<CC_TYPE>`: choose which custom controller backend to use, reboot required.   
```

File: dev/source/docs/copter-adding-custom-controller.rst
Comment (Hwurzburg) [inline dev/source/docs/copter-adding-custom-controller.rst:31]:
```suggestion
-  :ref:`CC_AXIS_MASK<CC_AXIS_MASK>`:  choose which axis to use custom controller output   
```

File: dev/source/docs/copter-adding-custom-controller.rst
Comment (Hwurzburg) [inline dev/source/docs/copter-adding-custom-controller.rst:53]:
```suggestion
The attitude target and rate target are also made equal to the current attitude and gyro rate to make the error signal grow from zero in ``relax_attitude_controllers`` function. In order to avoid impulse input to the controller, the target resetting is not performed when the feedforward is disabled by setting :ref:`ATC_RATE_FF_ENAB<ATC_RATE_FF_ENAB>`  parameter to 0.
```

File: dev/source/docs/copter-adding-custom-controller.rst
Comment (Hwurzburg) [inline dev/source/docs/copter-adding-custom-controller.rst:75]:
```suggestion
- Empty backend  set :ref:`CC_TYPE<CC_TYPE>`= 1
```

File: dev/source/docs/copter-adding-custom-controller.rst
Comment (Hwurzburg) [inline dev/source/docs/copter-adding-custom-controller.rst:76]:
```suggestion
- PID backend  set :ref:`CC_TYPE<CC_TYPE>`= 2
```

File: dev/source/docs/copter-adding-custom-controller.rst
Comment (Hwurzburg) [inline dev/source/docs/copter-adding-custom-controller.rst:79]:
```suggestion
Empty Controller -  :ref:`CC_TYPE<CC_TYPE>`= 1
```

File: dev/source/docs/copter-adding-custom-controller.rst
Comment (Hwurzburg) [inline dev/source/docs/copter-adding-custom-controller.rst:86]:
```suggestion
PID Controller - :ref:`CC_TYPE<CC_TYPE>`= 2
```

File: dev/source/docs/copter-adding-custom-controller.rst
Comment (Hwurzburg) [inline dev/source/docs/copter-adding-custom-controller.rst:98]:
```suggestion
**Step #1:**  Compile and run the default SITL model. In the GCS, choose the custom controller type, assign axis mask and set which RC switch to activate the custom controller. Reboot autopilot. For example in mavproxy, using RC channel 6 as the enable/disable switch for the custom controller:
```

**Review by Hwurzburg — CHANGES_REQUESTED** [review]
just a few nits....thanks

File: dev/source/docs/copter-adding-custom-controller.rst
Comment (esaldiran) [reply dev/source/docs/copter-adding-custom-controller.rst]: Done.

File: dev/source/docs/copter-adding-custom-controller.rst
Comment (esaldiran) [reply dev/source/docs/copter-adding-custom-controller.rst:29]: Done.

File: dev/source/docs/copter-adding-custom-controller.rst
Comment (esaldiran) [reply dev/source/docs/copter-adding-custom-controller.rst:31]: Done.

File: dev/source/docs/copter-adding-custom-controller.rst
Comment (esaldiran) [reply dev/source/docs/copter-adding-custom-controller.rst:53]: Done.

File: dev/source/docs/copter-adding-custom-controller.rst
Comment (esaldiran) [reply dev/source/docs/copter-adding-custom-controller.rst:75]: Done.

File: dev/source/docs/copter-adding-custom-controller.rst
Comment (esaldiran) [reply dev/source/docs/copter-adding-custom-controller.rst:79]: Done.

File: dev/source/docs/copter-adding-custom-controller.rst
Comment (esaldiran) [reply dev/source/docs/copter-adding-custom-controller.rst:98]: Done.

File: dev/source/docs/copter-adding-custom-controller.rst
Comment (esaldiran) [reply dev/source/docs/copter-adding-custom-controller.rst:86]: Done.

File: dev/source/docs/copter-adding-custom-controller.rst
Comment (esaldiran) [reply dev/source/docs/copter-adding-custom-controller.rst:76]: Done.

**Comment by Hwurzburg** [comment]
@esaldiran I corrected grammar and spelling....rather than force it over your branch I just added the commit, so you could see the changes....just to allow you to see them easily...before I merge this, when you are finished making changes, I will squash the commits before merging...thanks again for the great docs...

**Comment by Hwurzburg** [comment]
and a couple of old build errors that I had fixed previously, somehow crept back in...fixed

**Comment by esaldiran** [comment]
Thank you for the spell check @Hwurzburg. Everything looks good to me.

**Pull request merged**
